src/main/kotlin/com/sourcegraph/cody/edit/FixupService.kt

```diff
@@ -0,0 +1,166 @@
+package com.sourcegraph.cody.edit
+
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.components.service
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.sourcegraph.cody.agent.CodyAgentService
+import com.sourcegraph.config.ConfigUtil.isCodyEnabled
+import com.sourcegraph.utils.CodyEditorUtil
+
+/** Controller for commands that allow the LLM to edit the code directly. */
+@Service(Service.Level.PROJECT)
+class FixupService(val project: Project) : Disposable {
+  private val logger = Logger.getInstance(FixupService::class.java)
+
+  // We only use this for multiplexing task updates from the Agent to concurrent sessions.
+  // TODO: Consider doing the multiplexing in CodyAgentClient instead.
+  private var activeSessions: MutableMap<String, FixupSession> = mutableMapOf()
+
+  private var lastSelectedModel = "GPT-3.5"
+
+  // Sessions for which we have not yet received a task ID, but may receive an edit anyway.
+  private var pendingSessions: MutableSet<FixupSession> = mutableSetOf()
+
+  // The last text the user typed in without saving it, for continuity.
+  private var lastPrompt: String = ""
+
+  init {
+    // JetBrains docs say avoid heavy lifting in the constructor, so pass to another thread.
+    backgroundThread {
+      CodyAgentService.withAgent(project) { agent ->
+        ApplicationManager.getApplication().invokeLater {
```

Comment:
That `invokeLater` seems to be unnecessery:
1) It won't have effect on the thread in which callbacks will be called.
2) You ensure proper thread for callbacks, e.g. in `editTaskDidUpdate` using `onEventThread`, so you are covered

Reply:
Great catch - was the result of some refactoring. Fixed.